client/components/phone-input/country-flag.jsx

```diff
@@ -3,70 +3,30 @@
 /**
  * External dependencies
  */
-
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { PureComponent } from 'react';
 import Gridicon from 'gridicons';
 
-/** Internal Dependencies */
-import Spinner from 'components/spinner';
-
-export default class extends React.Component {
+export default class extends PureComponent {
 	static displayName = 'PhoneInputCountryFlag';
 
 	static propTypes = {
 		countryCode: PropTypes.string.isRequired,
 	};
 
-	state = {
-		ready: false,
-		error: false,
-	};
-
-	componentDidUpdate( oldProps ) {
-		if ( this.props.countryCode && this.props.countryCode !== oldProps.countryCode ) {
-			this.setState( { ready: false, error: false } );
-		}
-	}
-
-	renderSpinner = () => {
-		if ( ( ! this.props.countryCode || ! this.state.ready ) && ! this.state.error ) {
-			return <Spinner size={ 16 } className="phone-input__flag-spinner" />;
-		}
-	};
-
-	handleImageLoad = () => {
-		this.setState( { ready: true, error: false } );
-	};
-
-	handleImageError = () => {
-		this.setState( { ready: false, error: true } );
-	};
-
-	renderFlag = () => {
-		const style = this.state.ready ? {} : { visibility: 'hidden' };
-
-		if ( this.props.countryCode ) {
-			if ( ! this.state.error ) {
-				return (
-					<img
-						onLoad={ this.handleImageLoad }
-						onError={ this.handleImageError }
-						src={ `/calypso/images/flags/${ this.props.countryCode }.svg` }
-						className="phone-input__flag-icon"
-						style={ style }
-					/>
-				);
-			}
-			return <Gridicon icon="globe" size={ 24 } className="phone-input__flag-icon" />;
-		}
-	};
-
 	render() {
+		const { countryCode } = this.props;
+		// don't confusingly split the import statement with its long string
+		// note that webpack will parse this code at compile-time and auto-generate
+		// a context for loading the SVGs. it will infer the file glob for all SVG
+		// files in the given path and copy them to the public directory
+		//
+		// prettier-ignore
+		const flagSvg = require( '../../../node_modules/flag-icon-css/flags/4x3/' + countryCode + '.svg' );
```

Comment:
Could we do this with `asyncRequire` instead of `require`? I would assume this is pulling every flag into the bundle and some of the flags are pretty large. If we split it up with `asyncRequire` or `import()`, webpack can combine the small ones into one physical chunk and keep the bigger ones in separate files.

Reply:
Hmmm, maybe I'm wrong... icfy doesn't seem to be reporting very large size diffs, at least not as large as I would expect.

Reply:
ah ok, file-loader doesn't work the way I thought it did. It doesn't pull the files into the bundle at all, it just copies the file to a new place with a new name and gives the bundle a string to load the file in an img src. see https://github.com/webpack-contrib/file-loader

Reply:
correct: this returns a URL and that will be more obvious once I commit some updates which handle the failing flags in other screens